trying to implement the (ZMQ_IMMEDIATE) option

khaldoun1 opened the pull request:
This PR is a attempt to implement the 0mq option (ZMQ_IMMEDIATE)
Im not sure 100% if I did it the correct way, but it was requested by [ronkorving](https://github.com/ronkorving).

Im not expert neither c++ nor v8, and I need this option to be implmented urgently, so if I did a mistake, I'll be thankful for you if you correct it and do it for me.

for testing I did the below:
```javascript
var zmq = require('zeromq.node');
var socket = zmq.socket('push');
socket.setsockopt('immediate', 1);
```
but I got the error
```
Error: Invalid argument
at Socket.setsockopt (/home/kj/Development/Node_Test/node_modules/zeromq.node/lib/index.js:222:13)
at initClient (/home/kj/Development/Node_Test/zClient.js:10:9)
at Object. (/home/kj/Development/Node_Test/zClient.js:42:2)
at Module._compile (module.js:456:26)
at Object.Module._extensions..js (module.js:474:10)
at Module.load (module.js:356:32)
at Function.Module._load (module.js:312:12)
at Function.Module.runMain (module.js:497:10)
at startup (node.js:119:16)
at node.js:906:3
```

ronkorving commented:
Could you add a unit test? It's fine for now if it fails, but it'll allow us all to see better what's broken in your PR. I understand you're in a hurry, but I do want to maintain a particular level of quality here.

Also add it to https://github.com/JustinTulloss/zeromq.node/blob/master/test/exports.js which tests the availability of the options (depending on zmq version).

In the mean time, if anyone has any input on what the problem may be, don't hesitate to comment. Thanks.

ronkorving commented on lib/index.js:
```diff
@@ -37,6 +37,7 @@ var types = exports.types = {
   , router: zmq.ZMQ_ROUTER
   , pair: zmq.ZMQ_PAIR
   , stream: zmq.ZMQ_STREAM
+  , immediate: zmq.ZMQ_IMMEDIATE
```
You've put this in `types` (these are socket types), and should sit in `opts` (these are the options).

ronkorving commented:
I would still like to encourage you to do something to fix this, so we can merge it. Else I'm going to have to close this and we may implement it at a later time. Any questions are welcome.

khaldoun1 commented:
Thank you, 
I'll do ASAP within 1 week, Im sorry for this delay but I was too busy the last days.

regards

khaldoun1 commented:
Hello ronkorving

I've just implement the option in the correct place, and added a unit test  fot testing the mentioned option also updated the export one., but I still doubt that I did it right way.
hoping this may help.

in the other hand you may help by answering the question below:
in the file binding.cc what does the numbers in the following sentances mean:

```
    opts_int.insert(47); // ZMQ_CURVE_SERVER
    opts_binary.insert(48); // ZMQ_CURVE_PUBLICKEY
    opts_binary.insert(49); // ZMQ_CURVE_SECRETKEY
    opts_binary.insert(50); // ZMQ_CURVE_SERVERKEY
    opts_binary.insert(55); // ZMQ_ZAP_DOMAIN
```
I'm asking because I used (56) for the option I aimed to implement, I pick it because it is not already used and I tried to follow the serial

kkoopa commented:
https://github.com/zeromq/libzmq/blob/master/include/zmq.h#L277

ronkorving commented:
So much for that :)

khaldoun1 commented:
Im so sorry for everything, I've cloned the latest version (2.11.1) and create a [new pull request](https://github.com/JustinTulloss/zeromq.node/pull/438)

ronkorving commented:
Ah great, good luck! :)
